api_app/api/routes/shared_services.py

```diff
@@ -93,6 +93,8 @@ async def patch_shared_service(shared_service_patch: ResourcePatch, response: Re
         raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=strings.ETAG_CONFLICT)
     except ValidationError as v:
         raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=v.message)
+    except ValueError as v:
```

Comment:
Why are all these added?

Reply:
I've added a friendly error which tells the requester when an upgrade fails on target template version existence check.